src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -492,17 +492,22 @@ private function getAutowiredReference(TypedReference $reference, bool $filterTy
             }
         }
 
-        if ($this->container->has($type) && !$this->container->findDefinition($type)->isAbstract()) {
+        if ($this->container->has($type) && $this->canDefinitionBeAutowired($this->container->findDefinition($type))) {
             return new TypedReference($type, $type, $reference->getInvalidBehavior());
         }
 
-        if (null !== ($alias = $this->getCombinedAlias($type)) && !$this->container->findDefinition($alias)->isAbstract()) {
+        if (null !== ($alias = $this->getCombinedAlias($type)) && $this->canDefinitionBeAutowired($this->container->findDefinition($alias))) {
             return new TypedReference($alias, $type, $reference->getInvalidBehavior());
         }
 
         return null;
     }
 
+    private function canDefinitionBeAutowired(Definition $definition): bool
+    {
+        return !$definition->isAbstract() && !$definition->hasTag('container.excluded');
+    }
```

Comment:
Nitpick: you can move the `findDefinition` in the new method.

```suggestion
    private function canDefinitionBeAutowired(string $id): bool
    {
        $definition = $this->container->findDefinition($id);

        return !$definition->isAbstract() && !$definition->hasTag('container.excluded');
    }
```